nomos-services/cryptarchia-consensus/src/mix/adapters/mock.rs

```diff
@@ -0,0 +1,78 @@
+use nomos_mix_service::{
+    backends::mock::{Mock, MockEvent, MockEventKind, MockMessage},
+    NetworkMsg, NetworkService,
+};
+use overwatch_rs::services::{relay::OutboundRelay, ServiceData};
+use tokio::sync::broadcast::{self, error::RecvError};
+use tokio_stream::{wrappers::BroadcastStream, StreamExt};
+
+use crate::mix::{BoxedStream, NetworkAdapter};
+
+const BUFFER_SIZE: usize = 64;
+
+#[derive(Clone)]
+pub struct MockAdapter {
+    network_relay: OutboundRelay<<NetworkService<Mock> as ServiceData>::Message>,
+    mixed_msgs: broadcast::Sender<Vec<u8>>,
+}
+
+#[async_trait::async_trait]
+impl NetworkAdapter for MockAdapter {
+    type Backend = Mock;
+
+    async fn new(
+        network_relay: OutboundRelay<<NetworkService<Self::Backend> as ServiceData>::Message>,
+    ) -> Self {
+        let relay = network_relay.clone();
+        let mixed_msgs = broadcast::Sender::new(BUFFER_SIZE);
+        let mixed_msgs_sender = mixed_msgs.clone();
+
+        tokio::spawn(async move {
```

Comment:
Same stuff for this task.

Also, are we using the mock version?

Reply:
Fixed in 14b8de118e5eda5e8d1c3a20631e3ea6eff4fd84

In consensus service, no. We're using the libp2p adapter.

This mock adapter is for [DA integration test](https://github.com/logos-co/nomos-node/pull/824/files#diff-9d5cf4b9aae836e92a04ce9c9f4f1aec0b317470c26fa25907ac680a95a17bfa). I decided to use this mock because of [the comment](https://github.com/logos-co/nomos-node/pull/822/files#diff-80df468bf4cbe900aecbf5af48340caef3d57e1de0387d3edef75dad46fd4743L1) (which I removed in #822), but now I think it may be okay to use the real libp2p adapter for DA integration test, because it is already using the real libp2p adapter for the network service. Let me try it.

Reply:
Just moved the mock mix backend. We don't need that. Now we use the real mix backend for DA integration tests
b48ce841c7707c9d3f6756a27bbb44f48d3ccb04